src/internal/connector/onedrive/item.go

```diff
@@ -25,21 +25,25 @@ const (
 	downloadURLKey = "@microsoft.graph.downloadUrl"
 )
 
+// generic drive item getter
```

Comment:
@ryanfkeepers - we need to add a unit test for the reader with a mock client that simulates the errors we hit. The refactor you did actually enables that pretty easily (🤞 )

Reply:
Sounds good.  I'm unaware of the specific errors, if there's a link it'd be good to record those in a follow-up issue.